main.py

```diff
@@ -114,21 +124,42 @@ def search_key():
                     start = True
                     break
 
-                elif exitRect.collidepoint(event.pos):
-                    pygame.quit()
-                    sys.exit()
-
+                
                 elif incTurnsRect.collidepoint(event.pos):
-                    if maxturns<8:
+                    if maxturns<upperLimitTurnCount:
                         maxturns=maxturns+1
 
                 elif decTurnsRect.collidepoint(event.pos):
-                    if maxturns>4:
+                    if maxturns>lowerLimitTurnCount:
                         maxturns=maxturns-1
 
+                elif incGridSizeRect.collidepoint(event.pos):
+                    if gridLength<6:
+                        gridLength=gridLength+1
+                        
+                        upperLimitTurnCount, lowerLimitTurnCount = getLimits(gridLength)
+
+                        print(upperLimitTurnCount)
```

Comment:
Make sure to remove debug print statements from hereon, these aren't needed.

Reply:
Sure @vineetjc , I forgot to account for this print statement, will remove them next time.